### DISCUSSION TITLE
Why does the CDK example include a UUID api?

### DISCUSSION THREAD

**Discussion by coffeedome** [aws-powertools/powertools-lambda-typescript#1659] [Q&A] [answered]
Why does the example include a class UuidApi extends Construct? (class UuidApi extends Construct {} section)? It seems to have nothing to do with the Items API in that same example. 

Link to example: https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/src/example-stack.ts#L110

**Comment by dreamorosi** [comment]
Hi @coffeedome, the goal of the examples in the repository is to showcase usage of the Powertools utilities with projects deployed via CDK or SAM.

The UUID API specifically is used here as a [standalone REST API](https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/src/example-stack.ts#L120-L126) that accepts an HTTP request and returns a new UUID. As part of the `UuidApi` construct, we also [store an SSM parameter](https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/src/example-stack.ts#L132-L135) with the url of the API.

This parameter is then later used by some of the functions (i.e. [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/functions/put-item.ts#L62) or [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/functions/get-by-id.ts#L59)) to [obtain UUIDs](https://github.com/aws-powertools/powertools-lambda-typescript/blob/4e6f662b244a941a911c1ed5973bef11d7610093/examples/cdk/functions/common/getUuid.ts#L6). This allows us to showcase how to use the Parameters utility in your functions.

I agree that this example is somewhat trite and that we could improve the example by showing how to retrieve something more useful from SSM / Secrets Manager / etc, however in this case we opted for keeping the example simple over usefulness.

I hope this answers your question.

**Comment by dreamorosi** [comment, marked as answer]
We have refactored the examples some time ago to remove the API mentioned above, also in part because of this discussion evidencing it was confusing.
